doc/SAI-Proposal-Packet-Trimming.md

```diff
@@ -0,0 +1,142 @@
+                                                                                   │                └─┴─┴─┴─┴────────┘                  
+     ┌──────────────┐                                                              │                                                    
+     │              │  ┌──────────────────────────────────────────────────────┐    │                ┌─┬─┬─┬─┬─┬─┬─┬─┬┐                  
+     │              │  │                                                      │    │                │ │ │ │ │ │ │ │ ││                  
+     │              │  │                                                      │    │     \   /      │ │ │ │ │ │ │ │ ││                  
+     │              │  │                                                      │    │      \ /       │ │ │ │ │ │ │ │ ││                  
+     │    Packet    │  │           Pipeilne                                   ┼────┼───────\────────► │ │ │ │ │ │ │ ││     Queue        
+     │              │  │                                                      │           / \       │ │ │ │ │ │ │ │ ││                  
+     │              │  │                                                      │          /   \      │ │ │ │ │ │ │ │ ││                  
+     │              │  └──────────────────────────────────────────────────────┘                     └─┴─┴─┴─┴─┴─┴─┴─┴┘                  
+     │              │                                                                                                                   
+     │              │                                                                                                                   
+     │              │                                                                                                                   
+     └──────────────┘                                                                                                                   
+```
+
+This feature assumes that forwarding tables are configured properly, and the original packet would be delivered to the destination successfully if not for the congestion.
+
+## Spec
+There is a tradeoff between trying to configure a higher threshold in a queue buffer profile and trimming the packet.
+
+If the user chooses to configure higher thresholds for queues, the probability of a drop on a particular queue is lower only if other ports are less congested at the moment.
+
+However, if all the ports are equally utilized, it makes sense to create a different buffer profile for these queues, with a stricter threshold to have more fairness in shared buffer.
+
+For that, we propose adding a new attribute to a buffer profile to allow configuring packet trimming on such stricter profiles:
+```
+/**
+ * @brief Enum defining queue actions in case of packet discard.
+ */
+typedef enum _sai_buffer_profile_packet_discard_action_t
```

Comment:
can we find a better word than "discard"? Essentially the packet is not really discarded. It just fails to pass the admission control and we need to discard or trim it,

Reply:
Changed to admission_fail